trieste/acquisition/optimizer.py

```diff
@@ -11,42 +11,47 @@
 # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 # See the License for the specific language governing permissions and
 # limitations under the License.
+r"""
+This module contains functionality for optimizing
+:data:`~trieste.acquisition.AcquisitionFunction`\ s over :class:`~trieste.space.SearchSpace`\ s.
+"""
+from __future__ import annotations
+
 from functools import singledispatch
-from typing import Callable
+from typing import Callable, TypeVar
 
 import gpflow
 import tensorflow as tf
 import tensorflow_probability as tfp
 
 from ..space import Box, DiscreteSearchSpace, SearchSpace
 from ..type import TensorType
+from .function import AcquisitionFunction
 
-TensorMapping = Callable[[TensorType], TensorType]
-
+SP = TypeVar("SP", bound=SearchSpace)
+""" Type variable bound to :class:`~trieste.space.SearchSpace`. """
 
-@singledispatch
-def optimize(space: SearchSpace, target_func: TensorMapping) -> TensorType:
-    """
-    Return the point in ``space`` (with shape S) that maximises the function ``target_func``, as the
-    single entry in a 1 by S tensor.
 
-    ``target_func`` must satisfy the following:
+AcquisitionOptimizer = Callable[[SP, AcquisitionFunction], TensorType]
+"""
+Type alias for a function that returns the single point that maximizes an acquisition function over
+a search space. For a search space with points of shape S, and acquisition function with input shape
+[...] + S output shape [..., 1], the :data:`AcquisitionOptimizer` return shape should be [1] + S.
+"""
 
```

Comment:
In BO, input shape is normally referred to as d (i.e dimension). Is there a reason why you haven't done this here?

Reply:
nope, will change

Reply:
ah, that's why - D is the dimension, but S is a shape, which could be [D] or something more complex if working with higher-rank input spaces